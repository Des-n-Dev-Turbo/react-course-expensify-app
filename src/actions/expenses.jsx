import { v4 as uuidv4 } from 'uuid';

//ADDEXPENSE
export const addExpense = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}) => ({
    type: 'ADD_EXPENSE',
    expenses: {
        id: uuidv4(),
        description,
        note,
        amount,
        createdAt
    }
});

//REMOVEEXPENSE
export const removeExpense = ({ id }) => ({
    type: 'REMOVE_EXPENSE',
    id
});

//EDITEXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates

});
