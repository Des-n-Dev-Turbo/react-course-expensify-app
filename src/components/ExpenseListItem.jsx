import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = ({ description, id, createdAt, amount }) => (
  <Link to={`/edit/${id}`} className='list-item'>
    <div>
      <h3 className='list-item__title'>{description}</h3>
      <p className='list-item__sub-title'>{moment(createdAt).format("MMM Do, YYYY")}</p>
    </div>
    <div>
      <h3 className='list-item__data'>{"â‚¹" + numeral(amount).format("0,0.00")}</h3>
    </div>
  </Link>
);

export default ExpenseListItem;
