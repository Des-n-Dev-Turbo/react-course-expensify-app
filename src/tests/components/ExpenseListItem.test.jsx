import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Should Render the expense data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[2]}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});