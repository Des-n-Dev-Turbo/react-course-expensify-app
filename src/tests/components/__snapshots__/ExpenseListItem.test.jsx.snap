// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render the expense data 1`] = `
<div>
  <Link
    to="/edit/3"
  >
    <h4>
      Credit Card
    </h4>
  </Link>
  <p>
    4008.63
     - 
    Jan 5th, 1970
  </p>
</div>
`;
